compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -404,6 +404,7 @@ pub enum ValidationErrorKind<'tcx> {
     InvalidBool { value: String },
     InvalidChar { value: String },
     InvalidFnPtr { value: String },
+    GeneratorLayoutAccess { ty: Ty<'tcx>, generators: &'tcx List<Ty<'tcx>> },
```

Comment:
"ValidationErrorKind" is for when a value doesn't satisfy its validity invariant. This is unrelated, please don't put that here.

Sound more like an `InvalidProgram` error to me. Or ideally the error variant is only added in const_eval::error, so we structurally know it will not occur in Miri.